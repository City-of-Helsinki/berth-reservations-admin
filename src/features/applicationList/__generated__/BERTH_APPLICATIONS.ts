/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ApplicationStatus, LeaseStatus } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: BERTH_APPLICATIONS
// ====================================================

export interface BERTH_APPLICATIONS_berthApplications_edges_node_customer {
  __typename: "ProfileNode";
  id: string;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_berthSwitch_reason {
  __typename: "BerthSwitchReasonType";
  title: string | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_berthSwitch {
  __typename: "BerthSwitchType";
  berthNumber: string;
  harbor: string;
  harborName: string;
  id: string;
  pier: string;
  reason: BERTH_APPLICATIONS_berthApplications_edges_node_berthSwitch_reason | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_order {
  __typename: "OrderNode";
  id: string;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties_harbor_properties {
  __typename: "HarborProperties";
  name: string | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties_harbor {
  __typename: "HarborNode";
  id: string;
  properties: BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties_harbor_properties | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties {
  __typename: "PierProperties";
  identifier: string;
  harbor: BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties_harbor;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier {
  __typename: "PierNode";
  properties: BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier_properties | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth {
  __typename: "BerthNode";
  number: string;
  pier: BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth_pier;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_lease {
  __typename: "BerthLeaseNode";
  id: string;
  status: LeaseStatus;
  order: BERTH_APPLICATIONS_berthApplications_edges_node_lease_order | null;
  berth: BERTH_APPLICATIONS_berthApplications_edges_node_lease_berth;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node_harborChoices {
  __typename: "HarborChoiceType";
  harbor: string;
  priority: number;
  harborName: string;
}

export interface BERTH_APPLICATIONS_berthApplications_edges_node {
  __typename: "BerthApplicationNode";
  id: string;
  firstName: string;
  lastName: string;
  customer: BERTH_APPLICATIONS_berthApplications_edges_node_customer | null;
  berthSwitch: BERTH_APPLICATIONS_berthApplications_edges_node_berthSwitch | null;
  email: string;
  createdAt: any;
  municipality: string;
  boatType: string | null;
  boatRegistrationNumber: string;
  boatWidth: number;
  boatLength: number;
  boatDraught: number | null;
  boatWeight: number | null;
  boatName: string;
  boatModel: string;
  accessibilityRequired: boolean;
  status: ApplicationStatus;
  lease: BERTH_APPLICATIONS_berthApplications_edges_node_lease | null;
  harborChoices: (BERTH_APPLICATIONS_berthApplications_edges_node_harborChoices | null)[] | null;
}

export interface BERTH_APPLICATIONS_berthApplications_edges {
  __typename: "BerthApplicationNodeEdge";
  node: BERTH_APPLICATIONS_berthApplications_edges_node | null;
}

export interface BERTH_APPLICATIONS_berthApplications {
  __typename: "BerthApplicationNodeConnection";
  count: number;
  edges: (BERTH_APPLICATIONS_berthApplications_edges | null)[];
}

export interface BERTH_APPLICATIONS_boatTypes {
  __typename: "BoatTypeType";
  id: string;
  name: string | null;
}

export interface BERTH_APPLICATIONS {
  berthApplications: BERTH_APPLICATIONS_berthApplications | null;
  boatTypes: BERTH_APPLICATIONS_boatTypes[] | null;
}

export interface BERTH_APPLICATIONSVariables {
  first: number;
  after?: string | null;
  switchApplications?: boolean | null;
  orderBy?: string | null;
  statuses?: (ApplicationStatus | null)[] | null;
}
