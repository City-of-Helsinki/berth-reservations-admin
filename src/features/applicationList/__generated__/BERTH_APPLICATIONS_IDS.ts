/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ApplicationStatus } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: BERTH_APPLICATIONS_IDS
// ====================================================

export interface BERTH_APPLICATIONS_IDS_berthApplications_pageInfo {
  __typename: "PageInfo";
  hasNextPage: boolean;
  endCursor: string | null;
}

export interface BERTH_APPLICATIONS_IDS_berthApplications_edges_node {
  __typename: "BerthApplicationNode";
  id: string;
}

export interface BERTH_APPLICATIONS_IDS_berthApplications_edges {
  __typename: "BerthApplicationNodeEdge";
  node: BERTH_APPLICATIONS_IDS_berthApplications_edges_node | null;
}

export interface BERTH_APPLICATIONS_IDS_berthApplications {
  __typename: "BerthApplicationNodeConnection";
  count: number;
  pageInfo: BERTH_APPLICATIONS_IDS_berthApplications_pageInfo;
  edges: (BERTH_APPLICATIONS_IDS_berthApplications_edges | null)[];
}

export interface BERTH_APPLICATIONS_IDS {
  berthApplications: BERTH_APPLICATIONS_IDS_berthApplications | null;
}

export interface BERTH_APPLICATIONS_IDSVariables {
  first: number;
  after?: string | null;
  applicationCode?: boolean | null;
  switchApplications?: boolean | null;
  orderBy?: string | null;
  statuses?: (ApplicationStatus | null)[] | null;
  nameFilter?: string | null;
}
