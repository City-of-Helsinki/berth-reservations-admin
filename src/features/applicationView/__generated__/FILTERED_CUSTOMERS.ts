/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { OrganizationType } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: FILTERED_CUSTOMERS
// ====================================================

export interface FILTERED_CUSTOMERS_profiles_edges_node_primaryAddress {
  __typename: "AddressNode";
  address: string;
  postalCode: string;
  city: string;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_organization {
  __typename: "OrganizationNode";
  businessId: string;
  organizationType: OrganizationType;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor_properties {
  __typename: "HarborProperties";
  name: string | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor {
  __typename: "HarborNode";
  id: string;
  properties: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor_properties | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties {
  __typename: "PierProperties";
  identifier: string;
  harbor: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties_harbor;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier {
  __typename: "PierNode";
  properties: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier_properties | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth {
  __typename: "BerthNode";
  number: number;
  pier: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth_pier;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node {
  __typename: "BerthLeaseNode";
  berth: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node_berth;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges {
  __typename: "BerthLeaseNodeEdge";
  node: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges_node | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthLeases {
  __typename: "BerthLeaseNodeConnection";
  edges: (FILTERED_CUSTOMERS_profiles_edges_node_berthLeases_edges | null)[];
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node_berthSwitch {
  __typename: "BerthSwitchType";
  harborName: string;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node {
  __typename: "BerthApplicationNode";
  berthSwitch: FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node_berthSwitch | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges {
  __typename: "BerthApplicationNodeEdge";
  node: FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges_node | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges_node_berthApplications {
  __typename: "BerthApplicationNodeConnection";
  edges: (FILTERED_CUSTOMERS_profiles_edges_node_berthApplications_edges | null)[];
}

export interface FILTERED_CUSTOMERS_profiles_edges_node {
  __typename: "ProfileNode";
  id: string;
  firstName: string;
  lastName: string;
  primaryAddress: FILTERED_CUSTOMERS_profiles_edges_node_primaryAddress | null;
  organization: FILTERED_CUSTOMERS_profiles_edges_node_organization | null;
  berthLeases: FILTERED_CUSTOMERS_profiles_edges_node_berthLeases | null;
  berthApplications: FILTERED_CUSTOMERS_profiles_edges_node_berthApplications | null;
}

export interface FILTERED_CUSTOMERS_profiles_edges {
  __typename: "ProfileNodeEdge";
  node: FILTERED_CUSTOMERS_profiles_edges_node | null;
}

export interface FILTERED_CUSTOMERS_profiles {
  __typename: "ProfileNodeConnection";
  count: number | null;
  edges: (FILTERED_CUSTOMERS_profiles_edges | null)[];
}

export interface FILTERED_CUSTOMERS {
  profiles: FILTERED_CUSTOMERS_profiles | null;
}

export interface FILTERED_CUSTOMERSVariables {
  first: number;
  after?: string | null;
  firstName?: string | null;
  lastName?: string | null;
  email?: string | null;
  address?: string | null;
  orderBy?: string | null;
}
