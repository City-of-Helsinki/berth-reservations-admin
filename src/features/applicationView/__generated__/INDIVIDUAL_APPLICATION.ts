/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { BerthApplicationLanguage, InvoicingType, CustomerGroup, Language, ApplicationStatus, BerthMooringType, ProductServiceType, AdditionalProductType, PriceUnits, PeriodType } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: INDIVIDUAL_APPLICATION
// ====================================================

export interface INDIVIDUAL_APPLICATION_berthApplication_customer_organization {
  __typename: "OrganizationNode";
  id: string;
  address: string;
  businessId: string;
  city: string;
  name: string;
  postalCode: string;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_customer_primaryAddress {
  __typename: "AddressNode";
  id: string;
  address: string;
  postalCode: string;
  city: string;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_customer_primaryEmail {
  __typename: "EmailNode";
  id: string;
  email: string;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_customer_primaryPhone {
  __typename: "PhoneNode";
  id: string;
  phone: string | null;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_customer {
  __typename: "ProfileNode";
  comment: string | null;
  firstName: string;
  invoicingType: InvoicingType | null;
  lastName: string;
  id: string;
  customerGroup: CustomerGroup | null;
  organization: INDIVIDUAL_APPLICATION_berthApplication_customer_organization | null;
  primaryAddress: INDIVIDUAL_APPLICATION_berthApplication_customer_primaryAddress | null;
  primaryEmail: INDIVIDUAL_APPLICATION_berthApplication_customer_primaryEmail | null;
  primaryPhone: INDIVIDUAL_APPLICATION_berthApplication_customer_primaryPhone | null;
  language: Language | null;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_berthSwitch_reason {
  __typename: "BerthSwitchReasonType";
  title: string | null;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_berthSwitch {
  __typename: "BerthSwitchType";
  berthNumber: string;
  harbor: string;
  harborName: string;
  id: string;
  pier: string;
  reason: INDIVIDUAL_APPLICATION_berthApplication_berthSwitch_reason | null;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_harborChoices {
  __typename: "HarborChoiceType";
  harbor: string;
  priority: number;
  harborName: string;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_berth_pier_properties_harbor_properties {
  __typename: "HarborProperties";
  name: string | null;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_berth_pier_properties_harbor {
  __typename: "HarborNode";
  id: string;
  properties: INDIVIDUAL_APPLICATION_berthApplication_lease_berth_pier_properties_harbor_properties | null;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_berth_pier_properties {
  __typename: "PierProperties";
  identifier: string;
  electricity: boolean;
  gate: boolean;
  lighting: boolean;
  mooring: boolean;
  wasteCollection: boolean;
  water: boolean;
  harbor: INDIVIDUAL_APPLICATION_berthApplication_lease_berth_pier_properties_harbor;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_berth_pier {
  __typename: "PierNode";
  properties: INDIVIDUAL_APPLICATION_berthApplication_lease_berth_pier_properties | null;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_berth {
  __typename: "BerthNode";
  depth: number | null;
  length: number;
  mooringType: BerthMooringType;
  width: number;
  comment: string;
  isAccessible: boolean | null;
  number: string;
  pier: INDIVIDUAL_APPLICATION_berthApplication_lease_berth_pier;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_customer_primaryEmail {
  __typename: "EmailNode";
  id: string;
  email: string;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_customer {
  __typename: "ProfileNode";
  id: string;
  primaryEmail: INDIVIDUAL_APPLICATION_berthApplication_lease_customer_primaryEmail | null;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_order_orderLines_edges_node_product {
  __typename: "AdditionalProductNode";
  id: string;
  service: ProductServiceType;
  productType: AdditionalProductType;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_order_orderLines_edges_node {
  __typename: "OrderLineNode";
  id: string;
  price: any;
  product: INDIVIDUAL_APPLICATION_berthApplication_lease_order_orderLines_edges_node_product | null;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_order_orderLines_edges {
  __typename: "OrderLineNodeEdge";
  node: INDIVIDUAL_APPLICATION_berthApplication_lease_order_orderLines_edges_node | null;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_order_orderLines {
  __typename: "OrderLineNodeConnection";
  edges: (INDIVIDUAL_APPLICATION_berthApplication_lease_order_orderLines_edges | null)[];
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease_order {
  __typename: "OrderNode";
  id: string;
  orderNumber: string;
  price: any;
  totalPrice: any;
  orderLines: INDIVIDUAL_APPLICATION_berthApplication_lease_order_orderLines;
}

export interface INDIVIDUAL_APPLICATION_berthApplication_lease {
  __typename: "BerthLeaseNode";
  id: string;
  berth: INDIVIDUAL_APPLICATION_berthApplication_lease_berth;
  customer: INDIVIDUAL_APPLICATION_berthApplication_lease_customer;
  order: INDIVIDUAL_APPLICATION_berthApplication_lease_order | null;
}

export interface INDIVIDUAL_APPLICATION_berthApplication {
  __typename: "BerthApplicationNode";
  id: string;
  firstName: string;
  lastName: string;
  address: string;
  municipality: string;
  zipCode: string;
  phoneNumber: string;
  email: string;
  businessId: string;
  companyName: string;
  language: BerthApplicationLanguage;
  customer: INDIVIDUAL_APPLICATION_berthApplication_customer | null;
  berthSwitch: INDIVIDUAL_APPLICATION_berthApplication_berthSwitch | null;
  createdAt: any;
  boatType: string | null;
  boatRegistrationNumber: string;
  boatWidth: number;
  boatLength: number;
  boatDraught: number | null;
  boatWeight: number | null;
  boatName: string;
  boatModel: string;
  accessibilityRequired: boolean;
  status: ApplicationStatus;
  harborChoices: (INDIVIDUAL_APPLICATION_berthApplication_harborChoices | null)[] | null;
  acceptBoatingNewsletter: boolean;
  acceptFitnessNews: boolean;
  acceptLibraryNews: boolean;
  acceptOtherCultureNews: boolean;
  applicationCode: string;
  lease: INDIVIDUAL_APPLICATION_berthApplication_lease | null;
}

export interface INDIVIDUAL_APPLICATION_boatTypes {
  __typename: "BoatTypeType";
  id: string;
  name: string | null;
}

export interface INDIVIDUAL_APPLICATION_additionalProducts_edges_node {
  __typename: "AdditionalProductNode";
  id: string;
  service: ProductServiceType;
  priceValue: any;
  priceUnit: PriceUnits;
  period: PeriodType;
  productType: AdditionalProductType;
}

export interface INDIVIDUAL_APPLICATION_additionalProducts_edges {
  __typename: "AdditionalProductNodeEdge";
  node: INDIVIDUAL_APPLICATION_additionalProducts_edges_node | null;
}

export interface INDIVIDUAL_APPLICATION_additionalProducts {
  __typename: "AdditionalProductNodeConnection";
  edges: (INDIVIDUAL_APPLICATION_additionalProducts_edges | null)[];
}

export interface INDIVIDUAL_APPLICATION {
  berthApplication: INDIVIDUAL_APPLICATION_berthApplication | null;
  boatTypes: INDIVIDUAL_APPLICATION_boatTypes[] | null;
  __typename: "Query";
  additionalProducts: INDIVIDUAL_APPLICATION_additionalProducts | null;
}

export interface INDIVIDUAL_APPLICATIONVariables {
  id: string;
}
