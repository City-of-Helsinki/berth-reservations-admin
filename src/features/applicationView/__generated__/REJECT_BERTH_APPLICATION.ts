/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { RejectBerthApplicationMutationInput } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL mutation operation: REJECT_BERTH_APPLICATION
// ====================================================

export interface REJECT_BERTH_APPLICATION_rejectBerthApplication {
  __typename: "RejectBerthApplicationMutationPayload";
  clientMutationId: string | null;
}

export interface REJECT_BERTH_APPLICATION {
  rejectBerthApplication: REJECT_BERTH_APPLICATION_rejectBerthApplication | null;
}

export interface REJECT_BERTH_APPLICATIONVariables {
  input: RejectBerthApplicationMutationInput;
}
