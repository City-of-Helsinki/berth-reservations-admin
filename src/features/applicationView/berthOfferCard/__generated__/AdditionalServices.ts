/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ProductServiceType, PriceUnits, PeriodType, AdditionalProductType } from "./../../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL fragment: AdditionalServices
// ====================================================

export interface AdditionalServices_additionalProducts_edges_node {
  __typename: "AdditionalProductNode";
  id: string;
  service: ProductServiceType;
  priceValue: any;
  priceUnit: PriceUnits;
  period: PeriodType;
  productType: AdditionalProductType;
}

export interface AdditionalServices_additionalProducts_edges {
  __typename: "AdditionalProductNodeEdge";
  node: AdditionalServices_additionalProducts_edges_node | null;
}

export interface AdditionalServices_additionalProducts {
  __typename: "AdditionalProductNodeConnection";
  edges: (AdditionalServices_additionalProducts_edges | null)[];
}

export interface AdditionalServices {
  __typename: "Query";
  additionalProducts: AdditionalServices_additionalProducts | null;
}
