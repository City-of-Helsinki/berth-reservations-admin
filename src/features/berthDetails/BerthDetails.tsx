import React from 'react';
import { useTranslation } from 'react-i18next';
import Text from '../../common/text/Text';
import Grid from '../../common/grid/Grid';
import InternalLink from '../../common/internalLink/InternalLink';
import Section from '../../common/section/Section';
import { formatDate } from '../../common/utils/format';
import styles from './berthDetails.module.scss';
import MaintenanceServicesPlaceholder from '../../common/maintenancePlaceholders/MaintenanceServicesPlaceholder';
import { LeaseStatus } from '../../@types/__generated__/globalTypes';

export interface Lease {
  customer: {
    id: string;
    firstName: string;
    lastName: string;
  };
  startDate: string;
  endDate: string;
  status: LeaseStatus;
  isActive: boolean;
}

export interface BerthDetailsProps {
  leases: Lease[];
  comment: string;
  onEdit?(): void;
}

const BerthDetails = ({ leases, comment, onEdit }: BerthDetailsProps) => {
  const { t, i18n } = useTranslation();

  const expiredLeasesElements = leases
    .filter((lease) => !lease.isActive && lease.status === LeaseStatus.PAID)
    .map(({ startDate, endDate, customer }, i) => {
      return (
        <div key={i}>
          <Text>{`${formatDate(startDate, i18n.language)} - ${formatDate(endDate, i18n.language)}`}</Text>{' '}
          <InternalLink to={`/customers/${customer.id}`}>
            {customer.firstName !== '' && customer.lastName !== ''
              ? `${customer.firstName} ${customer.lastName}`
              : t('common.emptyName')}
          </InternalLink>
        </div>
      );
    });

  return (
    <div className={styles.berthDetails}>
      <Grid colsCount={3} className={styles.grid}>
        <Section title={t('offer.previousLeases').toUpperCase()}>
          {expiredLeasesElements.length ? expiredLeasesElements : '-'}
        </Section>
        <Section title={t('common.terminology.comments').toUpperCase()} className={styles.comment}>
          <Text>{comment || '-'}</Text>
        </Section>
        <MaintenanceServicesPlaceholder onClick={onEdit} buttonText={t('common.edit')} />
      </Grid>
    </div>
  );
};

export default BerthDetails;
