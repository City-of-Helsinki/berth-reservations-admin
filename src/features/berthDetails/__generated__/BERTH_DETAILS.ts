/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { LeaseStatus } from "./../../../@types/__generated__/globalTypes";

// ====================================================
// GraphQL query operation: BERTH_DETAILS
// ====================================================

export interface BERTH_DETAILS_berth_leases_edges_node_customer {
  __typename: "ProfileNode";
  /**
   * The ID of the object
   */
  id: string;
  firstName: string;
  lastName: string;
}

export interface BERTH_DETAILS_berth_leases_edges_node {
  __typename: "BerthLeaseNode";
  id: string;
  customer: BERTH_DETAILS_berth_leases_edges_node_customer;
  status: LeaseStatus;
  startDate: any;
  endDate: any;
  isActive: boolean;
}

export interface BERTH_DETAILS_berth_leases_edges {
  __typename: "BerthLeaseNodeEdge";
  node: BERTH_DETAILS_berth_leases_edges_node | null;
}

export interface BERTH_DETAILS_berth_leases {
  __typename: "BerthLeaseNodeConnection";
  edges: (BERTH_DETAILS_berth_leases_edges | null)[];
}

export interface BERTH_DETAILS_berth {
  __typename: "BerthNode";
  id: string;
  comment: string;
  leases: BERTH_DETAILS_berth_leases | null;
}

export interface BERTH_DETAILS {
  berth: BERTH_DETAILS_berth | null;
}

export interface BERTH_DETAILSVariables {
  berthId: string;
}
