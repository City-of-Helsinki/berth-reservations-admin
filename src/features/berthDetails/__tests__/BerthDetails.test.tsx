import React from 'react';
import { shallow } from 'enzyme';
import { HashRouter } from 'react-router-dom';

import BerthDetails, { BerthDetailsProps } from '../BerthDetails';
import { LeaseStatus } from '../../../@types/__generated__/globalTypes';

const minimumProps: BerthDetailsProps = {
  leases: [],
  comment: '',
};

const testLeases: BerthDetailsProps['leases'] = [
  {
    customer: {
      id: '0',
      firstName: '',
      lastName: '',
    },
    startDate: '2018-01-01',
    endDate: '2018-12-31',
    status: LeaseStatus.PAID,
    isActive: false,
  },
  {
    customer: {
      id: '1',
      firstName: 'Testi',
      lastName: 'Testinen',
    },
    startDate: '2019-01-01',
    endDate: '2019-12-31',
    status: LeaseStatus.PAID,
    isActive: false,
  },
  {
    customer: {
      id: '1',
      firstName: 'Testi',
      lastName: 'Testinen',
    },
    startDate: '2020-01-01',
    endDate: '2020-12-01',
    status: LeaseStatus.PAID,
    isActive: true,
  },
];

describe('BerthDetails', () => {
  const getWrapper = (props = {}) =>
    shallow(
      <HashRouter>
        <BerthDetails {...minimumProps} {...props} />
      </HashRouter>
    );

  it('renders normally with minimum props', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('renders normally with all props', () => {
    const wrapper = getWrapper({
      comment: 'Comment',
      leases: testLeases,
      onEdit: jest.fn(),
    });

    expect(wrapper.render()).toMatchSnapshot();
  });
});
