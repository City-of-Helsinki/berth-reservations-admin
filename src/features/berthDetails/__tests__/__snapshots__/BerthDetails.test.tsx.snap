// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BerthDetails renders normally with all props 1`] = `
<div
  class="berthDetails"
>
  <div
    class="grid cols3 grid"
  >
    <article
      class="section"
    >
      <h4
        class="text standard h4 m title"
      >
        EDELLISET VUOKRALAISET
      </h4>
      <section
        class="body"
      >
        <div>
          <span
            class="text standard span"
          >
            01.01.2018 - 31.12.2018
          </span>
           
          <a
            class="internalLink brand"
            href="#/customers/0"
            title="Nimi puuttuu"
          >
            Nimi puuttuu
          </a>
        </div>
        <div>
          <span
            class="text standard span"
          >
            01.01.2019 - 31.12.2019
          </span>
           
          <a
            class="internalLink brand"
            href="#/customers/1"
            title="Testi Testinen"
          >
            Testi Testinen
          </a>
        </div>
      </section>
    </article>
    <article
      class="section comment"
    >
      <h4
        class="text standard h4 m title"
      >
        HUOMIOT
      </h4>
      <section
        class="body"
      >
        <span
          class="text standard span"
        >
          Comment
        </span>
      </section>
    </article>
    <article
      class="section"
    >
      <div
        class="sectionHeadWithButton"
      >
        <h4
          class="text standard h4 m title"
        >
          HUOLTOTYÖT (PLACEHOLDER)
        </h4>
        <button
          class="title"
          type="button"
        >
          <span
            class="text brand span"
          >
            Muokkaa
          </span>
        </button>
      </div>
      <section
        class="body"
      >
        <table>
          <tbody>
            <tr>
              <td>
                <a
                  class="internalLink brand"
                  href="#/"
                  title="123"
                >
                  123
                </a>
              </td>
              <td>
                Portin lukko rikki
              </td>
            </tr>
            <tr>
              <td>
                <a
                  class="internalLink brand"
                  href="#/"
                  title="456"
                >
                  456
                </a>
              </td>
              <td>
                Poiju irti
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </article>
  </div>
</div>
`;

exports[`BerthDetails renders normally with minimum props 1`] = `
<div
  class="berthDetails"
>
  <div
    class="grid cols3 grid"
  >
    <article
      class="section"
    >
      <h4
        class="text standard h4 m title"
      >
        EDELLISET VUOKRALAISET
      </h4>
      <section
        class="body"
      >
        -
      </section>
    </article>
    <article
      class="section comment"
    >
      <h4
        class="text standard h4 m title"
      >
        HUOMIOT
      </h4>
      <section
        class="body"
      >
        <span
          class="text standard span"
        >
          -
        </span>
      </section>
    </article>
    <article
      class="section"
    >
      <h4
        class="text standard h4 m title"
      >
        HUOLTOTYÖT (PLACEHOLDER)
      </h4>
      <section
        class="body"
      >
        <table>
          <tbody>
            <tr>
              <td>
                <a
                  class="internalLink brand"
                  href="#/"
                  title="123"
                >
                  123
                </a>
              </td>
              <td>
                Portin lukko rikki
              </td>
            </tr>
            <tr>
              <td>
                <a
                  class="internalLink brand"
                  href="#/"
                  title="456"
                >
                  456
                </a>
              </td>
              <td>
                Poiju irti
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </article>
  </div>
</div>
`;
